### PR TITLE
GitLab integration docs

### PR DESCRIPTION
**Pull request by bjsignalfx**

### CODE REVIEW

File: gitlab/docs/runner/histogram.gitlab_runner_autoscaling_machine_creation_duration_seconds_sum.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: gitlab_runner_autoscaling_machine_creation_duration_seconds_sum
+brief: Histogram of machine creation time.
+metric_type: histogram
```
Comment (barbara-sfx) [inline gitlab/docs/runner/histogram.gitlab_runner_autoscaling_machine_creation_duration_seconds_sum.md]: General comment - our only metric types are gauge, counter, and cumulative counter. I see histogram, summary, and maybe others

File: gitlab/docs/runner/gauge.go_memstats_next_gc_bytes.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: go_memstats_next_gc_bytes
+brief: Number of heap bytes when next garbage collection will take place.
```
Comment (barbara-sfx) [inline gitlab/docs/runner/gauge.go_memstats_next_gc_bytes.md]: Another general comment - Brief descriptions can be more brief in some places. They show up in a table in the product docs, so really long ones will make the table a lot bigger. Not a big thing, just if you have time.

File: gitlab/docs/runner/counter.gitlab_runner_request_concurrency_exceeded_total.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+title: gitlab_runner_request_concurrency_exceeded_total
+brief: Counter tracking exceeding of request concurrency.
```
Comment (barbara-sfx) [inline gitlab/docs/runner/counter.gitlab_runner_request_concurrency_exceeded_total.md]: Another general comment - for consistency, brief descriptions shouldn't have a period at the end.

File: gitlab/README.md [hunk 4]
```diff
@@ -166,16 +169,9 @@ Sample of built-in dashboard in SignalFx:
...
-
-Configuring GitLab by editing `/etc/gitlab/gitlab.rb` should be accompanied by running the command `gitlab-ctl reconfigure` in order for the changes to take effect.
-
-
-Nginx configuration by editing file `/var/opt/gitlab/nginx/conf/nginx-status.conf` for instance, should be accompanied by running command `gitlab-ctl restart`. Note that changes to the configuration file `/var/opt/gitlab/nginx/conf/nginx-status.conf` in particular are erased by subsequent runs of command `gitlab-ctl reconfigure` because `gitlab-ctl reconfigure` restores the original configuration file. 
-
 ### METRICS
 
-For documentation of the metrics and dimensions emitted:
+For documentation of the metrics and dimensions emitted per metrics endpoint:
```
Comment (barbara-sfx) [inline gitlab/README.md]: Please check with @asuresh4 for the standard language here. I think his script expects specific wording.

File: gitlab/docs/runner/counter.gitlab_runner_api_request_statuses_total.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+title: gitlab_runner_api_request_statuses_total
+brief: The total number of api requests, partitioned by runner, endpoint and status.
+metric_type: counter
+---
+### gitlab_runner_api_request_statuses_total
+
+The total number of api requests, partitioned by runner, endpoint and status.
```
Comment (barbara-sfx) [inline gitlab/docs/runner/counter.gitlab_runner_api_request_statuses_total.md]: api ==> API throughout (in running text, obviously, not in the titles etc.)

**Comment by barbara-sfx** [comment]
@bjsignalfx  Does this need API docs? If so, please open a DOCS jira for @jmalin-signalfx   with any info on what APIs will be affected, who will be his SME, etc. Thanks.

**Comment by molner** [comment]
> @bjsignalfx Does this need API docs? If so, please open a DOCS jira for @jmalin-signalfx with any info on what APIs will be affected, who will be his SME, etc. Thanks.

No API docs for this @barbara-sfx

**Comment by barbara-sfx** [comment]
I didn't look at every single one of the 1000+ files, but on a quick scan it looks like you at least addressed the blockers, so this looks OK to merge.

**Pull request merged**